core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -504,6 +509,22 @@ public User createAccount(String userName, String password) throws IOException {
         return user;
     }
 
+    /**
+     * Creates a new user account by registering a JBCrypt Hashed password with the user.
+     *
+     * @param userName The user's name
+     * @param hashedPassword A hashed password, must begin with <code>#jbcrypt:</code>
+     */
+    public User createAccountWithHashedPassword(String userName, String hashedPassword) throws IOException {
+        if (!PASSWORD_ENCODER.isPasswordHashed(hashedPassword)) {
```

Comment:
There's a potential timing attack or validation check to extrapolate what number of rounds were chosen for bcrypt. Not sure if that's a security flaw, though.

Reply:
I do not understand what you are describing here as a "new" timing attack in the sense that any attempt to authenticate allow the user to determine the log number of rounds.

Reply:
I think it's very theoretical. You can keep trying hashes to figure out what number of rounds were configured on the server. Combined with a data breach, that may be useful for trying to brute force the hashes, but that's not exactly that feasible presently. Perhaps it might be a good idea to enforce a minimum number of rounds?